src/prometheus-net.DotNetRuntime/DotNetEventListener.cs

```diff
@@ -24,32 +20,28 @@ internal DotNetEventListener(IEventSourceStatsCollector collector, Action<Except
             _collector = collector;
             _errorHandler = errorHandler;
             _enableDebugging = enableDebugging;
-            
+
             if (_enableDebugging)
             {
                 _eventTypeCounts ??= Metrics.CreateCounter($"dotnet_debug_events_total", "The total number of .NET diagnostic events processed", "collector_name", "event_source_name", "event_name");
                 _cpuConsumed ??= Metrics.CreateCounter("dotnet_debug_cpu_seconds_total", "The total CPU time consumed by processing .NET diagnostic events (does not include the CPU cost to generate the events)", "collector_name", "event_source_name", "event_name");
                 _nameSnakeCase = collector.GetType().Name.ToSnakeCase();
             }
-            
-            EnableEventSources(collector);
+            EventSourceCreated += OnEventSourceCreated;
```

Comment:
Probably simply overriding OnEventSourceCreated method could be simpler, you will not have to be bothered about unsubscribing at all.